Updates setup adpat owner doc with cli cmds (#261)
microsoft/cobalt: docs/GETTING_STARTED_ADD_PAT_OWNER.md

  * *Global administrator role* permissions in your Organization's Azure Active Directory tenant to setup service principals
    * If this is not allowed by your organization, step two and the Service Connection creation in step three will need to be completed by someone within your organization with this permission.

> If using the CLI commands please set the following variables:

| Variable | Sample Value | Description |
|----------|--------------|-------------|

> | `APP_DEVOPS_PROJECT_NAME` | `My Application` | The name of the project representing your application that could include both your Cobalt foundation in addition to any application code, pipelines, boards, or artifacts. |

Reviewer: ```suggestion
| `TEMPLATE_DEVOPS_PROJECT_NAME` | `My Application` | The name of the project representing your Cobalt template application that serves as your organization's advocated pattern for a specific template. |
```

> | `APPS_DEVOPS_INFRA_REPO_NAME` | `infrastructure` | The name of the repo that will be created in the application Azure DevOps project to host the Cobalt template. |

Reviewer: ```suggestion
| `TEMPLATE_DEVOPS_INFRA_REPO_NAME` | `az-hello-world` | The name of the repo that will be created in the application Azure DevOps project to host the Cobalt template. |
```

> | `APP_DEVOPS_INFRA_YML_PATH` | `devops/providers/azure-devops/templates/azure-pipelines.yml` | The path relative to the `APPS_DEVOPS_INFRA_REPO_NAME` root that contains the Cobalt template pipeline to be created for provisioning application resources. |

Reviewer: ```suggestion
| `TEMPLATE_DEVOPS_INFRA_YML_PATH` | `devops/providers/azure-devops/templates/azure-pipelines.yml` | The path relative to the `TEMPLATE_DEVOPS_INFRA_REPO_NAME` root that contains the Cobalt template pipeline to be created for testing and provisioning resources. |
```

> | `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `APP_DEVOPS_PROJECT_NAME` resides or will be created. |

Reviewer: ```suggestion
| `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `TEMPLATE_DEVOPS_PROJECT_NAME` resides or will be created. |
```

> | `GIT_SOURCE_URL` | `https://dev.azure.com/MyOrganization/MyProject/_git/MyCobaltTemplateRepo` | The Git clone URL for the repository hosting the Cobalt template on which your project will be built. |

Reviewer: ```suggestion
| `COBALT_SOURCE_URL` | `https://github.com/microsoft/cobalt.git` | The Git clone URL for Cobalt (containing all templates including the one to be targeted by this template repository) from which this Cobalt template repository will be sourced. |
```

Update these values for your environment and application based on the guidance in the table above.
```bash

> export APP_DEVOPS_PROJECT_NAME=""

Reviewer: ```suggestion
export TEMPLATE_DEVOPS_PROJECT_NAME=""
```

> export APP_DEVOPS_INFRA_REPO_NAME=""

Reviewer: ```suggestion
export TEMPLATE_DEVOPS_INFRA_REPO_NAME=""
```

> export APP_DEVOPS_INFRA_YML_PATH=""

Reviewer: ```suggestion
export TEMPLATE_DEVOPS_INFRA_YML_PATH=""
```

export DEFAULT_ORGANIZATION=""

> export GIT_SOURCE_URL=""

Reviewer: ```suggestion
export COBALT_SOURCE_URL=""
```

            ![New Project](https://user-images.githubusercontent.com/10041279/63442791-4f868600-c3f9-11e9-91f3-c959654f5a1c.png)

        * Select Create

>     

Reviewer: ```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

```

    ```
    az devops configure --defaults organization="$DEFAULT_ORGANIZATION"

>     az devops project create --name "$APP_DEVOPS_PROJECT_NAME" --source-control git --visibility private

Reviewer: ```suggestion
    az devops project create --name "$TEMPLATE_DEVOPS_PROJECT_NAME" --source-control git --visibility private
```

>     az devops configure -d project="$APP_DEVOPS_PROJECT_NAME"

Reviewer: ```suggestion
    az devops configure -d project="$TEMPLATE_DEVOPS_PROJECT_NAME"
```

            | Cobalt-AZ-ISO-Contoso | If the aim is to have a single repository represent a single Cobalt template, and thereafter, to have one repo per template, we recommend a name that reflects the Cobalt Template being deployed. In this naming example, the name assumes this repo will be dedicated to deploying the Cobalt *az-isolated-service-single-region* template |
            | Cobalt-Contoso | If the aim is to use a single repository as ground truth for all potential patterns across your organization, effectively having to manage a combination of Cobalt patterns from a single repo, it's recommended to stick with a name that matches the project name. |

>     ```

Reviewer: ```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

        ```
```

>     az repos create --name "$APP_DEVOPS_INFRA_REPO_NAME"

Reviewer: ```suggestion
    az repos create --name "$TEMPLATE_DEVOPS_INFRA_REPO_NAME"
```

>     az repos import create --git-url https://github.com/microsoft/cobalt.git --repository "$APP_DEVOPS_INFRA_REPO_NAME"

Reviewer: ```suggestion
    az repos import create --git-url $COBALT_SOURCE_URL --repository "$TEMPLATE_DEVOPS_INFRA_REPO_NAME"
```

            > NOTE: Azure Devops forces a run so expect this to fail. Future steps will resolve this problem.

>     ```

Reviewer: ```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

>     az pipelines create --name "$COBALT_PIPELINE_NAME" --repository "$APP_DEVOPS_INFRA_REPO_NAME" --branch master --repository-type tfsgit --yml-path $APP_DEVOPS_INFRA_YML_PATH --skip-run true

Reviewer: ```suggestion
    az pipelines create --name "$COBALT_PIPELINE_NAME" --repository "$TEMPLATE_DEVOPS_INFRA_REPO_NAME" --branch master --repository-type tfsgit --yml-path $TEMPLATE_DEVOPS_INFRA_YML_PATH --skip-run true
```

        * Once deployment for storage account is completed, go to the resource and visit the Blobs sub-menu
        * Click [+Container] then create a container name (ex. az-hw-remote-state-container or az-iso-remote-state-container) with private access

> ```

Reviewer: ```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions. Note this creates the Service Connection within Azure DevOps, but you will still need to setup the Service Principal:

```
```

> az devops service-endpoint azurerm create --azure-rm-subscription-id $SUBSCRIPTION_ID --azure-rm-subscription-name $SUBSCRIPTION_NAME --azure-rm-tenant-id $TENANT_ID --azure-rm-service-principal-id $SERVICE_PRIN_ID --name $SERVICE_CONN_NAME

Reviewer: ```suggestion
az devops service-endpoint azurerm create --azure-rm-subscription-id $SUBSCRIPTION_ID --azure-rm-subscription-name "$SUBSCRIPTION_NAME" --azure-rm-tenant-id $TENANT_ID --azure-rm-service-principal-id $SERVICE_PRIN_ID --name "$SERVICE_CONN_NAME"
```

    > Important: Every targeted environment specified within the build pipeline expects a
    > variable group specified with the naming convention `<ENVIRONMENT_NAME> Environment Variables`

>     ```

Reviewer: ```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

    # IMPORTANT: Replace these values as necessary to fit your environment.
    az pipelines variable-group create --authorize true --name "$COBALT_VAR_GROUP_INFRA" --variables \

>         AGENT_POOL='Hosted Ubuntu 1604' \

Reviewer 2: Can we parameterize this to `$AGENT_POOL_NAME`?

        ARM_PROVIDER_STRICT=true \
        BUILD_ARTIFACT_NAME='drop' \
        BUILD_ARTIFACT_PATH_ALIAS='artifact' \
        GO_VERSION='1.12.5' \
        PIPELINE_ROOT_DIR='devops/providers/azure-devops/templates/infrastructure' \
        REMOTE_STATE_CONTAINER='BACKENDSTATECONTAINERNAME' \
        SCRIPTS_DIR='scripts' \
        TEST_HARNESS_DIR='test-harness/' \

>         TF_DEPLOYMENT_TEMPLATE_ROOT='infra/templates/az-hello-world' \

Reviewer 2: ```suggestion
        TF_DEPLOYMENT_TEMPLATE_ROOT='infra/templates/$TEMPLATE_DEVOPS_INFRA_REPO_NAME' \
```

            | `REMOTE_STATE_ACCOUNT` | `<AZURE_STORAGE_ACCOUNT_NAME>` | The storage container account name created in a previous step that is used to manage the state of this deployment pipeline. The storage Account is shared among all non-prod deployment stages. |
            | `SERVICE_CONNECTION_NAME` | ex. Cobalt Deployment Administrator-`<TenantName>` | The custom name of the service connection configured in a previous Azure Devops step that establishes a connection between the Service Principal and the Azure subscription that it's permissioned for. |

>     ```

Reviewer: ```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

    # IMPORTANT: Replace these values as necessary to fit your environment.
    DEVINT_VAR_GROUP="DevInt $COBALT_VAR_GROUP_ENV_SUFFIX"

>     az pipelines variable-group create --authorize true --name $DEVINT_VAR_GROUP --variables \

Reviewer: ```suggestion
    az pipelines variable-group create --authorize true --name "$DEVINT_VAR_GROUP" --variables \
```

        * Save the build pipeline

>     ```

Reviewer: ```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

>     az pipelines show --name "Cobalt-Hello-World-Pipeline" -o json > builddef.json

Reviewer: ```suggestion
    az pipelines show --name "$COBALT_PIPELINE_NAME" -o json > builddef.json
```

>     PIPELINE_ID=$(az pipelines show --name "Cobalt-Hello-World-Pipeline" --query id)

Reviewer: ```suggestion
    PIPELINE_ID=$(az pipelines show --name "$COBALT_PIPELINE_NAME" --query id)
```

    ```

> 

Reviewer: ```suggestion

Execute the list command to find the Variable Group IDs created earlier. Make note of the IDs as they will need to be added to the build pipeline definition.

    ```bash
    az pipelines variable-group list
    ```
```

    For the workaround, you'll be manually editing the builddef.json file to add the variable group references. At the end of the file, you should see the line `"variableGroups" : null`. Replace the value with the following, replacing the variable group ID placeholders (`0`) with those from the above command for the Infrastructure Pipeline Variables group and DevInt Environment Variables group:
    ```bash
    "variableGroups": [
        { "id": 0 },
        { "id": 0 }
    ],
    ```

    ```

>     az devops invoke --http-method PUT --area build --resource definitions --route-parameters project="TestCobalt" definitionId=10 --in-file builddef.json

Reviewer: ```suggestion
    az devops invoke --http-method PUT --area build --resource definitions --route-parameters project="$TEMPLATE_DEVOPS_PROJECT_NAME" definitionId=$PIPELINE_ID --in-file builddef.json
```
